### PR TITLE
Add name-template flag for installation that allows the user to speci…

### PR DESCRIPTION
**Pull request by vaikas**
…fy a template used for release name generation

If I understood the issue correctly, something like this?
#975

### CODE REVIEW

File: cmd/helm/install.go [hunk 1]
```diff
@@ -17,13 +17,17 @@ limitations under the License.
...
 import (
+	"bytes"
 	"fmt"
 	"io"
 	"io/ioutil"
 	"os"
 	"path/filepath"
 	"strings"
 
+	"html/template"
```
Comment (technosophos) [inline cmd/helm/install.go]: I think we want `text/template`, though I'm not sure that it will make a difference in this case.

Comment (vaikas-google) [inline cmd/helm/install.go]: I tried to switch to it and it doesn't work with the funcmaps. But, asides from that, is this roughly what you had in mind for the feature?

Comment (technosophos) [inline cmd/helm/install.go]: Oh, yeah... you have to use `sprig.TxtFuncMap()`. Forgot about that.

Otherwise, yes, this is _exactly_ what I had in mind.

Comment (vaikas-google) [inline cmd/helm/install.go]: Ah, great. That works. Thanks, PTAL.

**Comment by vaikas-google** [comment]
@viglesiasce Could you take a look and see if this would help your use case?

**Comment by technosophos** [comment]
Manual testing:

```console
$  helm install --name-template "technosophos-{{randAlpha 6 | lower}}" docs/examples/alpine
final name: technosophos-hxaawt
technosophos-hxaawt
```

I think we could remove `final-name: YYY` from the output, but it all works exactly as envisioned.

**Comment by vaikas-google** [comment]
Thanks!!! Vic, did I misunderstand how you wanted the name output format to look. I can remove the final name but just lmk. If I don't hear soon, I'll just submit and we can omit it later.

**Comment by viglesiasce** [comment]
Hey @vaikas-google! Sorry for the late response. Yeah I'd prefer either:

1. Just the derived app name
2. A way to configure what output format comes out json, yaml, simple (default like 1 above)

Maybe I can use this as my first PR attempt.

**Pull request merged**